pkgs/by-name/sd/sdl3/package.nix

```
@@ -0,0 +1,183 @@
+    ++ lib.optional dbusSupport dbus
+    ++ lib.optionals drmSupport [
+      libdrm
+      mesa # libgbm
+    ]
+    ++ lib.optional jackSupport libjack2
+    ++ lib.optional libdecorSupport libdecor
+    ++ lib.optional libudevSupport systemdLibs
+    ++ lib.optional openglSupport libGL
+    ++ lib.optional pipewireSupport pipewire
+    ++ lib.optional pulseaudioSupport libpulseaudio
+    ++ lib.optional sndioSupport sndio
+    ++ lib.optionals waylandSupport [
+      wayland
+      wayland-scanner
+    ]
+    ++ lib.optionals x11Support [
+      libxkbcommon
+      xorg.libX11
+      xorg.libXScrnSaver
+      xorg.libXcursor
+      xorg.libXext
+      xorg.libXfixes
+      xorg.libXi
+      xorg.libXrandr
+    ];
+
+  propagatedBuildInputs = finalAttrs.dlopenPropagatedBuildInputs;
+
+  dlopenPropagatedBuildInputs =
+    lib.optional (openglSupport && !stdenv.hostPlatform.isDarwin) libGL
+    ++ lib.optional x11Support xorg.libX11;
+
+  cmakeFlags = [
+    (lib.cmakeBool "SDL_ALSA" alsaSupport)
+    (lib.cmakeBool "SDL_DBUS" dbusSupport)
+    (lib.cmakeBool "SDL_IBUS" ibusSupport)
+    (lib.cmakeBool "SDL_JACK" jackSupport)
+    (lib.cmakeBool "SDL_KMSDRM" drmSupport)
+    (lib.cmakeBool "SDL_LIBUDEV" libudevSupport)
+    (lib.cmakeBool "SDL_OPENGL" openglSupport)
+    (lib.cmakeBool "SDL_PIPEWIRE" pipewireSupport)
+    (lib.cmakeBool "SDL_PULSEAUDIO" pulseaudioSupport)
+    (lib.cmakeBool "SDL_SNDIO" sndioSupport)
+    (lib.cmakeBool "SDL_TEST_LIBRARY" testSupport)
+    (lib.cmakeBool "SDL_WAYLAND" waylandSupport)
+    (lib.cmakeBool "SDL_WAYLAND_LIBDECOR" libdecorSupport)
+    (lib.cmakeBool "SDL_X11" x11Support)
+
+    (lib.cmakeBool "SDL_TESTS" finalAttrs.finalPackage.doCheck)
+  ];
+
+  # testautomation & testautomation-no-simd fail
+  # doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+
+  # See comment below. We actually *do* need these RPATH entries
+  dontPatchELF = true;
+
+  env = {
+    # Many dependencies are not directly linked to, but dlopen()'d at runtime. Adding them to the RPATH
+    # helps them be found
+    NIX_LDFLAGS =
+      lib.optionalString (stdenv.hostPlatform.extensions.sharedLibrary == ".so")
+        "-rpath ${
+          lib.makeLibraryPath (finalAttrs.dlopenBuildInputs ++ finalAttrs.dlopenPropagatedBuildInputs)
+        }";
+  };
+
+  passthru = {
+    tests.pkg-config = testers.hasPkgConfigModules { package = finalAttrs.finalPackage; };
+
+    # TODO: Use upstream's prerelease tags when they start making them again
+    # https://github.com/libsdl-org/SDL/issues/10367
+    updateScript = nix-update-script {
+      extraArgs = [
+        "--version-regex"
+        "'preview-(.*)'"
+      ];
+    };
+  };
+
+  meta = {
+    description = "Cross-platform development library (Preview version)";
+    homepage = "https://libsdl.org";
+    license = lib.licenses.zlib;
+    maintainers = with lib.maintainers; [ getchoo ];
+    pkgConfigModules = [ "sdl3" ];
+  };
```

`meta.platforms` is needed, `stdenv.mkDerivation` doesn't default this to anything meaningful like the `rustPlatform.buildRustPackage`: https://github.com/NixOS/nixpkgs/blob/06d251bedc53d1591f0ce796cc86c56a5e48323a/pkgs/README.md#new-packages

```suggestion
    platforms = lib.platforms.unix;
  };
```